### PR TITLE
DRAFT Add task instance event log table

### PR DESCRIPTION
**Pull request by dstandish**
draft!

### CODE REVIEW

File: airflow/api_connexion/schemas/event_log_schema.py [hunk 1]
```diff
@@ -43,17 +44,41 @@ class Meta:
...
+
+
 class EventLogCollection(NamedTuple):
     """List of import errors with metadata."""
 
     event_logs: list[Log]
     total_entries: int
 
+class TaskInstanceEventLogCollection(NamedTuple):
+    """List of import errors with metadata."""
```
Comment (vincbeck) [inline airflow/api_connexion/schemas/event_log_schema.py]: Copy pasta

File: airflow/providers/amazon/aws/executors/ecs/ecs_executor.py [hunk 2]
```diff
@@ -385,18 +385,22 @@ def attempt_task_runs(self):
                     )
                     self.pending_tasks.append(ecs_task)
                 else:
-                    self.send_message_to_task_logs(
-                        logging.ERROR,
-                        "ECS task %s has failed a maximum of %s times. Marking as failed. Reasons: %s",
-                        task_key,
+                    reasons_str = ", ".join(failure_reasons)
+                    self.log.error(
+                        "ECS task %s has failed %s times. Marking as failed. Reasons: %s" % task_key,
```
Comment (vincbeck) [inline airflow/providers/amazon/aws/executors/ecs/ecs_executor.py]:
```suggestion
                        "ECS task %s has failed %s times. Marking as failed. Reasons: %s",
```

And pass `task_key` as first parameter

**Review by vincbeck — COMMENTED** [review]
Looks good overall! Should we also deprecate/remove `TaskContextLogger `?

**Comment by dstandish** [comment]
closing in favor of #40739

**Pull request closed without merging**